crates/turbopack-core/src/resolve/mod.rs

```diff
@@ -1943,6 +1942,36 @@ async fn resolve_relative_request(
                 .collect(),
         ));
 
+        if options_value.enable_js_ts_rewriting {
+            let mut rewritten_path = path_pattern.clone();
+            let rewritten_path_modified =
+                rewritten_path.replace_final_constants(&|c: &RcStr| -> Option<Pattern> {
+                    let result = match c.rsplit_once(".") {
+                        Some((base, "js")) => Some((
+                            base,
+                            vec![
+                                Pattern::Constant(".ts".into()),
+                                Pattern::Constant(".tsx".into()),
+                            ],
+                        )),
+                        Some((base, "mjs")) => Some((base, vec![Pattern::Constant(".mts".into())])),
+                        Some((base, "cjs")) => Some((base, vec![Pattern::Constant(".cts".into())])),
+                        _ => None,
+                    };
+                    result.map(|(base, replacement)| {
+                        Pattern::Concatenation(vec![
+                            Pattern::Constant(base.into()),
+                            Pattern::Alternatives(replacement),
+                        ])
+                    })
+                });
+
+            if rewritten_path_modified {
+                // Prepend the rewritten pattern to give it higher priority
+                new_path = Pattern::Alternatives(vec![rewritten_path, new_path])
```

Comment:
Instead of doing that you can just keep the original extension in the list for Alternatives above